custom/samveg/case_importer/validators.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+
+from django.utils.translation import ugettext as _
+
+from corehq.util.dates import get_previous_month_date_range
+from custom.samveg.case_importer.exceptions import UnexpectedFileError
+from custom.samveg.const import (
+    MANDATORY_COLUMNS,
+    RCH_BENEFICIARY_IDENTIFIER,
+    RCH_MANDATORY_COLUMNS,
+    SNCU_BENEFICIARY_IDENTIFIER,
+    SNCU_MANDATORY_COLUMNS,
+)
+
+
+class BaseValidator:
+    @classmethod
+    def run(cls, *args, **kwargs):
+        raise NotImplementedError
+
+    @classmethod
+    def _get_latest_call_value_and_number(cls, raw_row):
```

Comment:
Sorry yes, I already knew it would be called later in https://github.com/dimagi/commcare-hq/pull/30928/files#diff-e2ad39182718855af1cb8edaa1ad795f29b9c18d7acca6e5db02fb8791eb57b8R118, so I had it in the base class.